core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -280,9 +285,53 @@ protected ClassLoader createClassLoader(List<File> paths, ClassLoader parent, At
     }
 
     /**
+     * Get the list of all plugins that have ever been detached from Jenkins core.
```

Comment:
Btw, shouldn't that text be somehow introduced below in the DetachPlugin class javadoc?